### PR TITLE
Fix sparse format of `kron`

### PR DESCRIPTION
**Pull request by toslunar**
Follow scipy/scipy#13100

### CODE REVIEW

**Comment by toslunar** [comment]
The `cupy_pr` CI should have SciPy 1.6 tests.  Blocked by chainer/chainer-test#630.

File: tests/cupyx_tests/scipy_tests/sparse_tests/test_construct.py [hunk 1]
```diff
@@ -401,7 +401,7 @@ def test_diags_non_square(self, xp, sp):
     'arrA': arrs,
     'arrB': arrs,
 }))
-@testing.with_requires('scipy')
+@testing.with_requires('scipy>=1.6')
```
Comment (leofang) [inline tests/cupyx_tests/scipy_tests/sparse_tests/test_construct.py:404]: Would this requirement be too strong? This method already exists before SciPy 1.6. Perhaps we can add the requirement at the method level?

Comment (emcastillo) [reply tests/cupyx_tests/scipy_tests/sparse_tests/test_construct.py:404]: I think adding here it is fine, there is only one test and currently, it won't pass for scipy<1.6

Comment (toslunar) [reply tests/cupyx_tests/scipy_tests/sparse_tests/test_construct.py:404]:
```python
if testing.scipy_satisfies('<1.6'):  # similar to testing.numpy_satisfies
    kron = kron.A
```

is an option.  But I justify the current fix as a kind of white-box test.  CuPy's `kron` does not directly refer `cupy.sparse`, i.e. the dependency `scipy>=1.6` is only for the testing purpose.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Review by emcastillo — APPROVED** [review]

**Pull request merged**